### ISSUE TITLE
Cluster Connections: can we cancel randomness and keep the given order?

### ISSUE THREAD

**Issue by ea167** [guidesmiths/rascal#65]
Hello,

Thank you for your great wrapper lib.
I'm trying to setup a RabbitMQ cluster with an active/passive behaviour for high-availability, where I only want to connect to the main node except if it fails.

Is there a way to disable the randomness in the vhost > connections configuration?
I would like the client to systematically connect in the given order.

Thank you!

**Comment by cressie176** [comment]
Hi @ea167,

This is not possible at the moment, but I agree it's a good feature. 

There are a few options for implementing this...

### New `connectionStrategy` vhost level property
```
{
  "vhosts": {
    "v1": {
      "connectionStrategy": "fixed",
      "connections": [
         "amqp://user:password@host1/v1",
         "amqp://user:password@host2/v1",
         "amqp://user:password@host3/v1",
      ]
    },
    "v2": {
      "connectionStrategy": "random",
      "connections": [
         "amqp://user:password@host1/v2",
         "amqp://user:password@host2/v2",
         "amqp://user:password@host3/v2",
      ]
    }
  }
}
```
The connectionStrategy would default to "random" for backwards compatibility. When "fixed" it maintain the order of connections in the array. 

#### Pros
* Simplicity
* Little change
* Supports different strategies per vhost

#### Cons 
* The strategy and url order must be repeated for each vhost if consistent behaviour is desired (partially mitigated by using rascal's `defaults` feature).
* Specifying a connection strategy, but only having a single connection doesn't make sense

Thoughts?

**Comment by ea167** [comment]
Brilliant Stephen @cressie176! Would be perfect!

I guess it only impacts `lib/config/configure.js` > `getConnectionIndex`,
where you can test `connectionStrategy` and branch between `Math.random()` and a simple sequence increment.

**Comment by cressie176** [comment]
Yes, thanks what I'm thinking.

**Comment by cressie176** [comment]
Published as rascal@4.7.0

**Comment by ea167** [comment]
Bravo Stephen @cressie176! Amazing! 👏👏👏
